pkg/model/apps.go

```diff
@@ -56,28 +57,33 @@ func (r *Apps) AddApps(item v1alpha1.Application) {
 func (r *Apps) AddAppsFromSet(item v1alpha1.ApplicationSet) {
 	tracking := item.Annotations["argocd.argoproj.io/tracking-id"]
 	if tracking != "" {
-		tracking = strings.Split(tracking, ":")[0]
-	} else {
+		// Parse the tracking ID which has format:
+		// applicationName:argoproj.io/ApplicationSet:namespace/applicationSetName
+		parts := strings.Split(tracking, ":")
+		if len(parts) == 3 && strings.Contains(parts[1], "ApplicationSet") {
+			// Use the parent Application name as tracking
+			tracking = parts[0]
+		}
+	}
+
+	if tracking == "" {
+		// Fallback to owner references if no tracking annotation
 		references := item.OwnerReferences
 		if len(references) != 0 {
 			tracking = references[0].Name
 		} else {
-			println(item.GetName(), " has no refrence")
+			println(item.GetName(), " has no tracking annotation argocd.argoproj.io/tracking-id")
 		}
 	}
-	destName := item.Spec.Template.Spec.Destination.Name
-	//Use Server when name is not used
-	if destName == "" {
-		destName = item.Spec.Template.Spec.Destination.Server
-	}
 
 	r.Apps = append(r.Apps, App{
 		Name:      item.Name,
 		ManagedBy: tracking,
-		Cluster:   destName,
+		Cluster:   "in-cluster",
```

Comment:
I don't know if this is working correctly when renaming `in-cluster`.
We have renamed our `in-cluster` because it was confusing for users.

https://github.com/argoproj/argo-cd/discussions/13219